unidad_3/mpi/ejercicio_3/ejercicio_3_paredes.cpp

```diff
@@ -0,0 +1,54 @@
+/* escribir un programa paralelo que encadene el envío y recepción de un mensaje, mensaje sera el rango (identificador) del proceso que envía.
+
+   el proceso sero debe enviar su nro. de rango al segundo, el segundo recibirá dicho mensaje y enviará otro al tercero, y así sucesivamente para todos los procesos lanzados.
+
+   Todo proceso que reciba un mensaje debe imprimirlo de la siguiente manera:
+   
+   ```
+   "Soy el proceso X y he recibido M",
+   ```
+
+   siendo X el rango del proceso y M el mensaje recibido. */
+
+    #include <mpi.h>
+    #include <stdio.h>
+    #include <stdlib.h>
+    #include <string.h>
+
+    int main(int argc, char *argv[])
+    {
+        int rank, size, tag = 0;
+        MPI_Status status;
+        MPI_Init(&argc, &argv);
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+        MPI_Comm_size(MPI_COMM_WORLD, &size);
+        int i;
+        int mensaje;
+        for (i = 0; i < size; i++)
+        {
+            if (rank == i)
```

Comment:
cual seria la funcion de este `for`? 

en este caso, cada proceso hace un for de N iteraciones, pero ejecuta la iteracion que unicamente coincide con su numero de rango. 

acordate que este codigo se ejecuta por cada proceso

Reply:
que les parece @Denisdif @FacuRodriJ @jorferr89

Reply:
Todo el código se ejecuta por cada proceso. Directamente se debe ir al if de la línea 31. 
Otra cosa, el if rank == 0 solo debe ejecutar la función MPI_Send, y no el MPI_Recv... el proceso con id 0 debe enviar un mensaje